drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTableModellerViewImpl.java

```diff
@@ -0,0 +1,1042 @@
+                for ( ConditionCol52 c : conditions ) {
+                    HorizontalPanel hp = new HorizontalPanel();
+                    hp.setStylePrimaryName( GuidedDecisionTableResources.INSTANCE.css().patternConditionSectionHeader() );
+                    if ( isEditable ) {
+                        hp.add( removeCondition( c ) );
+                    }
+                    hp.add( editCondition( p,
+                                           c ) );
+                    SmallLabel conditionLabel = makeColumnLabel( c );
+                    hp.add( conditionLabel );
+                    conditionsPanel.add( hp );
+                }
+
+            } else if ( conditionColumn instanceof BRLConditionColumn ) {
+                BRLConditionColumn brl = (BRLConditionColumn) conditionColumn;
+
+                HorizontalPanel patternHeaderPanel = new HorizontalPanel();
+                patternHeaderPanel.setStylePrimaryName( GuidedDecisionTableResources.INSTANCE.css().patternSectionHeader() );
+                HorizontalPanel patternPanel = new HorizontalPanel();
+                if ( isEditable ) {
+                    patternPanel.add( removeCondition( brl ) );
+                }
+                patternPanel.add( editCondition( brl ) );
+                Label patternLabel = makePatternLabel( brl );
+                patternPanel.add( patternLabel );
+                patternHeaderPanel.add( patternPanel );
+                patternsPanel.add( patternHeaderPanel );
+            }
+
+        }
+    }
+
+    private Label makePatternLabel( final Pattern52 p ) {
+        StringBuilder patternLabel = new StringBuilder();
+        String factType = p.getFactType();
+        String boundName = p.getBoundName();
+        if ( factType != null && factType.length() > 0 ) {
+            if ( p.isNegated() ) {
+                patternLabel.append( GuidedDecisionTableConstants.INSTANCE.negatedPattern() ).append( " " ).append( factType );
+            } else {
+                patternLabel.append( factType ).append( " [" ).append( boundName ).append( "]" );
+            }
+        }
+        return new Label( patternLabel.toString() );
+    }
+
+    private Label makePatternLabel( final BRLConditionColumn brl ) {
+        StringBuilder sb = new StringBuilder();
+        sb.append( brl.getHeader() );
+        return new Label( sb.toString() );
+    }
+
+    private SmallLabel makeColumnLabel( final ConditionCol52 cc ) {
+        StringBuilder sb = new StringBuilder();
+        if ( cc.isBound() ) {
+            sb.append( cc.getBinding() );
+            sb.append( " : " );
+        }
+        sb.append( cc.getHeader() );
+        SmallLabel label = new SmallLabel( sb.toString() );
+        if ( cc.isHideColumn() ) {
+            label.setStylePrimaryName( GuidedDecisionTableResources.INSTANCE.css().columnLabelHidden() );
+        }
+        return label;
+    }
+
+    private Widget editCondition( final Pattern52 origPattern,
```

Comment:
Perhaps another spot for a bit of 'DRY'ing-up..? (for the overloaded editCondition(..))